MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs

```diff
@@ -2829,17 +2841,25 @@ private void strncpy()
             }
         }
 
+        /// <summary>
+        ///     Registers a new Text Variable that will be used by the module
+        /// </summary>
         private void register_textvar()
         {
-            var textPointer = GetParameterPointer(0);
+            var name = GetParameterString(0, true);
             var functionPointer = GetParameterPointer(2);
 
-            var textBytes = Module.Memory.GetString(textPointer, true);
+            var newTextVar = new TextvarStruct(name, functionPointer);
+            var newTextVarOffset = Module.Memory.GetWord("NTVARS") * TextvarStruct.Size;
 
-            Module.TextVariables.Add(Encoding.ASCII.GetString(textBytes), functionPointer);
+            //Save
+            Module.Memory.SetArray(Module.Memory.GetVariablePointer("TXTVARS") + newTextVarOffset, newTextVar.Data);
+
+            //Increment
+            Module.Memory.SetWord("NTVARS", (ushort) (Module.Memory.GetWord("NTVARS") + 1));
```

Comment:
reuse newTextVarOffset?

Reply:
newTextVarOffset calculates the offset in the array by multiplying `NTVARS` by the struct size. `NTVARS` contains just hte number of variables. So we increment it here by one.